lib/score_tracker.rb

```diff
@@ -4,27 +4,29 @@ class ScoreTracker
   DRAW_SCORE = 0
   INCOMPLETE_GAME = -1000000
 
-  def initialize(player_1_token, player_2_token, win_checker)
-    @win_checker = win_checker
-    @player_scores = initialize_player_scores(player_1_token, player_2_token)
-    @player_tokens = [player_1_token, player_2_token]
+  def initialize(board)
+    rules = TicTacToeRules.new(board.get_side_length)
+    @win_checker = WinChecker.new(board, rules)
+    tokens = board.get_player_tokens
+    @player_scores = initialize_player_scores(tokens[0], tokens[1])
+    @player_tokens = [tokens[0], tokens[1]]
   end
 
   def get_player_score(token)
     @player_scores[token]
   end
 
   def is_game_finished?
-    player_1_token = @player_tokens[0]
+    player_a_token = @player_tokens[0]
 
-    @player_scores[player_1_token] != INCOMPLETE_GAME
+    @player_scores[player_a_token] != INCOMPLETE_GAME
   end
 
   def has_either_player_won?
-    player_1_token = @player_tokens[0]
+    player_a_token = @player_tokens[0]
 
-    @player_scores[player_1_token] == WIN_SCORE ||
-    @player_scores[player_1_token] == LOSS_SCORE
+    @player_scores[player_a_token] == WIN_SCORE ||
+    @player_scores[player_a_token] == LOSS_SCORE
```

Comment:
Indent here to make it clear these two lines are one expression.

Reply:
Indent added.